### PR TITLE
feat(wallet): implement bridge to aurora button

### PR DESCRIPTION
**Pull request by muliswilliam**
Resolves https://github.com/brave/brave-browser/issues/24377

## Submitter Checklist:

- [x] I confirm that no security/privacy review is not needed
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Click on an asset in portfolio list for example Ethereum
2. On the asset details page, if the asset support aurora bridging, you will see a button with the text `Bridge to Aurora`
3. Click on the button, a modal should appear with bridging details.
4. Click on `Open rainbowbridge.app`. A new tab should open and load `https://rainbowbridge.app`

https://user-images.githubusercontent.com/48117473/182663673-444c9ecf-1dba-416d-88cc-23d7025b0bbd.mov

### CODE REVIEW

File: components/brave_wallet_ui/components/desktop/views/portfolio/portfolio-asset.tsx [hunk 1]
```diff
@@ -295,6 +309,24 @@ export const PortfolioAsset = () => {
 
   const onNftDetailsLoad = React.useCallback(() => setNftIframeLoaded(true), [])
 
+  const onToggleShowBridgeToAurora = React.useCallback(() => {
+    if (bridgeToAuroraWarningShown) {
+      onOpenRainbowAppClick()
+    } else {
+      setShowBridgeToAuroraModal(prevShoModal => !prevShoModal)
+      localStorage.setItem(bridgeToAuroraWarningShownKey, 'true')
+    }
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/views/portfolio/portfolio-asset.tsx:318]: This one has a bug where trying to close the modal will open the bridge URL

I recommend creating dedicated open/close functions

**Review by josheleonard — CHANGES_REQUESTED** [review]

File: components/brave_wallet_ui/components/desktop/popup-modals/bridge-to-aurora-modal/bridge-to-aurora-modal.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    <PopupModal
+      title=''
+      onClose={onClose}
+      width={modalWidth}
+    >
+      <StyledWrapper>
+        <Title>{getLocale('braveWalletAuroraModalTitle')}</Title>
+        <Description>{getLocale('braveWalletAuroraModalDescription')}</Description>
+        <LearnMoreLink
+          target='_blank'
```
[the remark this answers is not in the document]
Comment (muliswilliam) [reply components/brave_wallet_ui/components/desktop/popup-modals/bridge-to-aurora-modal/bridge-to-aurora-modal.tsx:40]: @josheleonard Fixed here: https://github.com/brave/brave-core/commit/d71f2c820be86a9b832ad69ac6551ff9e9abd6fa

File: components/brave_wallet_ui/components/desktop/views/portfolio/portfolio-asset.tsx (hunk 1 above)
Comment (muliswilliam) [reply components/brave_wallet_ui/components/desktop/views/portfolio/portfolio-asset.tsx]: @josheleonard Good catch . Fixed here 9e6a282b2a0f5906ac5039fb9b311f474f81756f

File: components/brave_wallet_ui/components/desktop/views/portfolio/portfolio-asset.tsx [hunk 3]
```diff
@@ -295,6 +309,28 @@ export const PortfolioAsset = () => {
 
   const onNftDetailsLoad = React.useCallback(() => setNftIframeLoaded(true), [])
 
+  const onBridgeToAuroraButton = React.useCallback(() => {
+    if (bridgeToAuroraWarningShown) {
+      onOpenRainbowAppClick()
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/views/portfolio/portfolio-asset.tsx:314]: `onOpenRainbowAppClick` used before it was defined

**Review by josheleonard — CHANGES_REQUESTED** [review]

File: components/brave_wallet_ui/components/desktop/views/portfolio/portfolio-asset.tsx (hunk 3 above)
Comment (muliswilliam) [reply components/brave_wallet_ui/components/desktop/views/portfolio/portfolio-asset.tsx:323]: @josheleonard Fixed here: 4665ea288ab16b67804ca03bf67df711af362a58

**Review by josheleonard — APPROVED** [review]

**Comment by srirambv** [comment]
Verification passed on 

Brave | 1.44.26 Chromium: 104.0.5112.81 (Official Build) nightly (64-bit) 
-- | --
Revision | `5b7b76419d50f583022568b6764b630f6ddc9208-refs/branch-heads/5112@{#1309}`
OS | Windows 11 Version 21H2 (Build 22000.795)

- Verified steps from PR 
- Verified for supported asset, there is a button `Bridge to Aurora` on the asset details screen
- Verified clicking on the button shows a modal about bridging details
- Verified `Learn more` opens a new tab page and loads `https://ethereum.org/en/bridges/#bridge-risk`
- Verified the modal is shown only once for the first time. Subsequent attempts doesn't show the modal

https://user-images.githubusercontent.com/17010094/183375559-cfd61600-fa2c-4f36-b8ba-d39b584c2415.mp4

**Pull request merged**